enhancements/console/workflow-guides.md

```diff
@@ -0,0 +1,175 @@
+
+### Implementation Details
+
+1. GitHub will be used creating and storing the Workflow Guides. GitHub provides rich markdown options, including images and videos, which will come handy in the process of creating a Workflow Guide. Since we have storage limits in ETCD, it's for the best to use a technology that both storest the content and provides rich markdown. Also by using GitHub we solve the issue with  Workflow Guides versioning.
+2. For linking the OpenShift Console and the Workflow Guides `ConsoleLinks` CRD will be used. To suite the need following extension of the `ConsoleLinks` is needed:
+   1. Add additional `ConsoleLinkLocation` type to the `ConsoleLink` [API](https://github.com/openshift/api/blob/master/console/v1/types_console_link.go#L65-L74) named `WorkflowGuide`.
+   2. If `ConsoleLinkLocation` type is set to `WorkflowGuide`, it will be possible to define a `WorkflowGuideSpec` with following fields:
+      * Section - string - Defines area of interest, e.g. `operators`.
+      * Admin - bool - Defines if the Workflow Guide is cluster admin oriented (user has to have cluster admin permisions).
+      * Version - string - Defines the version of the `WorkflowGuide`
+      * **Extenstion:**: ResourceCheckName - string - Defines the name of the `ResourceCheck` CR.
+
+3. RedHat supported Workflow Guides will be places in new `openshift/workflow-guides` repository. Each root folder in the repository will be named after area of interest (e.g. top items from the left navigation menu -  Operators, Workloads, Serverless, ..., Administration).
+4. The `openshift/workflow-guides` repository will also contain a `manifests/` root folder that will contain all yaml for all the official Workflow Guides `ConsoleLinks` CRs. All the manifests will be copied into the `console-operator` image upon it's build. Alternativly there could be a small `workflow-guides-operator` that will be:
+   1. Contain all the `ConsoleLinks` CRs from the `manifests/` root folder.
+   2. Responsile for removing old `ConsoleLinks` CRs (based on the cluster version) after a cluster update, so the `ConsoleLink` CR points to a proper version of the `Workflow Guide` on GitHub.
+5. The `openshift/workflow-guides` repository will contain a script that would be responsible for updating all the `ConsoleLinks` CRs to point to the current version.
+6. All Workflow Guides will be listed available in a separate page that will be accessible from Help Menu.
+   ![help-menu](https://raw.githubusercontent.com/jhadvig/images/master/help-menu.png)
+7. For different areas of interest, the links for Workflow Guides will need to be handles separately:
+   * Operators - CVS of the operator that has a Workflow Guide published will contain `console.openshift.io/workflow-guide: <console-link-name>` annotation. Upon the operator installation an "Workflow Guide" link will appear next to the `Install` button.
```

Comment:
Is the intent for console to list CSVs in the current namespace in addition to `ConsoleLink` to build one list of guides?

Reply:
no, this is only related to the `Operator Hub` page where we use the operator's CSV for rendering it in a tile. The CSV could contain mentioned annotation in order to link to a particular ConsoleLink CR.
Im only mention ing the list of the Guides since the UX team was proposing some kind of their list and that could actually be a separate page where user could list though all the Guides